risc0/zkp/prove/prove.cpp

```diff
@@ -256,22 +195,37 @@ BufferU32 prove(const std::string& elfFile, MemoryHandler& io) {
   iop.commit(hashU);
 
   // Set the mix mix value
-  Fp4 mix;
-  for (size_t i = 0; i < kMixMixGlobalSize; i++) {
-    mix.elems[i] = Fp::random(iop);
-    exec.context.globals[kMixMixGlobalOffset + i] = mix.elems[i];
-  }
+  Fp4 mix = Fp4::random(iop);
+  ;
   LOG(1, "Mix = " << mix);
 
   // Do the coefficent mixing
-  auto combos = AccelSlice<Fp4>::allocate(size * (kComboCount + 1));
-  mixPolyCoeffsAccel(combos,
-                     codeGroup.getCoeffs(),
-                     dataGroup.getCoeffs(),
-                     accumGroup.getCoeffs(),
-                     checkGroup.getCoeffs(),
-                     AccelSlice<Fp>::copy(exec.context.globals, kGlobalSize),
-                     size);
+  // Begin by making a zeroed output buffer
+  auto combos = AccelSlice<Fp4>::copy(std::vector<Fp4>(size * (kComboCount + 1)));
+  Fp4 curMix(1);
+  auto mixGroup = [&](RegisterGroup id, PolyGroup& pg) {
```

Comment:
Wondering if this could just be a separate function? Maybe there's a ton of things that would need to be captured otherwise?